src/transform-sdk/js/main.cc

```diff
@@ -315,6 +315,323 @@ std::expected<qjs::value, qjs::exception> make_write_event(
     return write_event;
 }
 
+class schema_registry_client {
+public:
+    explicit schema_registry_client(
+      std::unique_ptr<redpanda::sr::schema_registry_client> client)
+      : _client(std::move(client)) {}
+    schema_registry_client(const schema_registry_client&) = delete;
+    schema_registry_client& operator=(const schema_registry_client&) = delete;
+    schema_registry_client(schema_registry_client&&) = default;
+    schema_registry_client& operator=(schema_registry_client&&) = default;
+    ~schema_registry_client() = default;
+
+    std::expected<qjs::value, qjs::exception>
+    lookup_schema_by_id(JSContext* ctx, std::span<qjs::value> params) {
+        if (params.size() != 1) [[unlikely]] {
+            return std::unexpected(qjs::exception::make(
+              ctx,
+              std::format(
+                "invalid number of parameters to "
+                "SchemaRegistryClient.lookupSchemaById, got: {}, "
+                "expected: 1",
+                params.size())));
+        }
+        auto& param = params.front();
+        if (!param.is_number()) [[unlikely]] {
+            return std::unexpected(qjs::exception::make(
+              ctx,
+              "expected integer for 'id' param to "
+              "SchemaRegistryClient.lookupSchemaById"));
+        }
+
+        return _client
+          ->lookup_schema_by_id(redpanda::sr::schema_id{
+            static_cast<redpanda::sr::schema_id::type>(param.as_integer())})
+          .transform_error([ctx](std::error_code err) {
+              return qjs::exception::make(
+                ctx, std::format("schema lookup failed: {}", err.message()));
+          })
+          .and_then([ctx](redpanda::sr::schema schema) {
```

Comment:
forgot to run the linter locally, but this is sort of an interesting case. i think the value of the upstream expected gets delivered as an rvalue ref in these pipelined cases, so the guidance is to take either by value or by rvalue ref. 

need to give a little thought to whether silencing the linter or massaging ref semantics is more correct here.